### DISCUSSION TITLE
Create offer based on budget of activities

### DISCUSSION THREAD

**Discussion by Marty** [kimai/kimai#1855] [General]
**Is your feature request related to a problem? Please describe.**
For a new project I'm about to acquire, I create activities and put estimated budgets on them. I can not export the activities and their budgets to create an offer for my clients.

**Describe the solution you'd like**
I'd like to be able to export the activities of one project along with their budgets to create an offer document.
This may, in a first and easy step, be possible just by extending the available template variables. A more elegant solution would be to have a separate menu point under which all offers are stored. These offers could then be converted to actual invoices.

**Describe alternatives you've considered**
I found no way of exporting all the activities of a project along with their budgets.
One workaround could be to create fictional time entries and use them together with a custom invoice template to "fake" an offer by creating an actual invoice. But I then had to delete these time entries as they would corrupt the statistics.

**Additional context**
If that's not already possible, maybe you could point me in the right direction, where to start, so that I may create a plugin for that.

**Comment by kevinpapst** [comment]
Hm, you could use the API to get a list of all activities from the client and create an offer from those.
Should be possible to create that as a one pager (there is a demo in the documentation how to use the API from a HTML page).

A nicer solution would be a plugin, that lets you select the the customer (and project(s) (and activities)) and then generate the offer from there. This could be stored to the database for future reference.

Not sure what you need, as I write offers very rarely myself and have no need for it ... so I never thought a feature like that.
